docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,236 @@
+.. _custom_constraints:
+
+Customize your Constraints
+==========================
+
+In some cases, you would like to construct your own constraint. You can
+use ``CustomConstraint`` to define your own logic on how you would like to
+apply it to your data. There are three main functions that you can create:
+
+- ``transform`` which is responsible for the forward pass when using ``transform`` strategy.
+- ``reverse_transform`` which defines how to reverse the transformation of the ``transform``.
+- ``is_valid`` which computes the invalid rows.
+
+Let's look at a demo dataset:
+
+.. ipython:: python
+    :okwarning:
+
+    from sdv.demo import load_tabular_demo
+
+    employees = load_tabular_demo()
+    employees
+
+The dataset defined in :ref:`_single_table_constraints` contains basic details about employees.
+We will use this dataset to demonstrate how you can create your own constraint. 
+
+
+Custom Constraints
+------------------
+
+Let's consider the following example where we wish to generate synthetic data and we want
+a particular column, age for example, to be even. We will define ``transform``, 
+``reverse_transform``, and ``is_valid`` methods to make our data satisfy our constraints.
+
+
+In this example, we would like to create a constraint that makes ``age`` even. We can
+create a ``transform`` function that makes all the ``age`` values even.
+
+.. ipython:: python
+    :okwarning:
+
+    def transform(table_data):
+        table_data['age'] = table_data['age'] * 2
+        return table_data
+
+We transform the column ``age`` to be even by multiplicating it with 2. We now define
+the ``reverse_transform`` as doing the opposite effect.
+
+.. ipython:: python
+    :okwarning:
+
+    def reverse_transform(table_data):
+        table_data['age'] = table_data['age'] / 2
+        return table_data
+
+Lastly, we write the ``is_valid`` function to assess whether the column is even or not.
+
+.. ipython:: python
+    :okwarning:
+
+    def is_valid(table_data):
```

Comment:
Doubting about whether to include the `is_valid` in the main example, or just move it to the `note` below as an alternative possibility.

Reply:
I have this doubt too. I think there is a branch for ``transform`` and ``reverse_transform``; another branch for ``is_valid`` but I think this distinction should be on a higher level. In other words, we should describe the essence of these functions in the `Constraints` page, specifically in the ``handling_strategy``. And here in ``CustomConstraint`` we link to the place.

Then I would keep all three functions here defined such that the signature of the function is very clear.

Reply:
I'm a bit concerned about this being confusing, so one may thing that it is necessary to implement the three functions every time. What do you think about creating a new section at the end for this instead?

We could cover the three styles using `transform` and `reverse_transform` only, and then add a new section at the end (e.g. titled "Can I write a CustomConstraint based on reject sampling?") that explains only that.
This section could just mention that if there is no transform/reverse transform implementation possible, the `is_valid` functions with the same signature can be passed instead, and then show the three implementations at once at the end.